examples/step-68/step-68.cc

```diff
@@ -0,0 +1,835 @@
+    const unsigned int particle_weight = 10000;
+
+    // This example does not use adaptive refinement, therefore every cell
+    // should have the status `CELL_PERSIST`. However this function can also
+    // be used to distribute load during refinement, therefore we consider
+    // refined or coarsened cells as well.
+    if (status == parallel::distributed::Triangulation<dim>::CELL_PERSIST ||
+        status == parallel::distributed::Triangulation<dim>::CELL_REFINE)
+      {
+        const unsigned int n_particles_in_cell =
+          particle_handler.n_particles_in_cell(cell);
+        return n_particles_in_cell * particle_weight;
+      }
+    else if (status == parallel::distributed::Triangulation<dim>::CELL_COARSEN)
+      {
+        unsigned int n_particles_in_cell = 0;
+
+        for (unsigned int child_index = 0; child_index < cell->n_children();
+             ++child_index)
+          n_particles_in_cell +=
+            particle_handler.n_particles_in_cell(cell->child(child_index));
+
+        return n_particles_in_cell * particle_weight;
+      }
+
+    Assert(false, ExcInternalError());
+    return 0;
+  }
+
+
+
+  // @sect4{Particles generation}
+
+  // This function generates the tracer particles and the background
+  // triangulation on which these particles evolve.
+  template <int dim>
+  void ParticleTracking<dim>::generate_particles()
+  {
+    // We create a hyper cube triangulation which we globally refine. This
+    // triangulation covers the full trajectory of the particles.
+    GridGenerator::hyper_cube(background_triangulation, 0, 1);
+    background_triangulation.refine_global(par.fluid_refinement);
+
+    // In order to consider the particles when repartitioning the triangulation
+    // the algorithm needs to know three things:
+    //
+    // 1. How much weight to assign to each cell (how many particles are in
+    // there);
+    // 2. How to pack the particles before shipping data around;
+    // 3. How to unpack the particles after repartitioning.
+    //
+    // We attach the correct functions to the signals inside
+    // parallel::distributed::Triangulation. These signal will be called every
+    // time the repartition() function is called. These connections only need to
+    // be created once, so we might as well have set them up in the constructor
+    // of this class, but for the purpose of this example we want to group the
+    // particle related instructions.
+    background_triangulation.signals.cell_weight.connect(
+      [&](
+        const typename parallel::distributed::Triangulation<dim>::cell_iterator
+          &cell,
+        const typename parallel::distributed::Triangulation<dim>::CellStatus
+          status) -> unsigned int { return this->cell_weight(cell, status); });
+
+    background_triangulation.signals.pre_distributed_repartition.connect(
+      std::bind(
+        &Particles::ParticleHandler<dim>::register_store_callback_function,
+        &particle_handler));
```

Comment:
```suggestion
    background_triangulation.signals.pre_distributed_repartition.connect(
      [this]() {this->particle_handler.register_store_callback_function(); });
```

Reply:
This works for the register_store_callback_function.
However, the register_load_callback_function expects an argument in the form of a serialization (see https://www.dealii.org/current/doxygen/deal.II/classParticles_1_1ParticleHandler.html#a9a79889c1ad3b569a405488d4bab8263)
How do I deal with this? Because 
`
    background_triangulation.signals.post_distributed_repartition.connect(
      [this]() {
        this->particle_handler.register_load_callback_function();
      });
`
does not work obviously since it is missing an argument. Sorry again for asking, but my experience with signals or lambdas is extremely limited.

Reply:
Cannot you simply write?
```cpp
    background_triangulation.signals.post_distributed_repartition.connect(
      [&](){this->particle_handler.register_load_callback_function(false)});
```

Reply:
Ok this seems to compile well. But now the code segfaults in parallel when partitioning. I'll try to debug and see what happens. Some change somehow must be broken things.